src/native/eventpipe/ep-sample-profiler.c

```diff
@@ -205,13 +205,44 @@ sample_profiler_enable (void)
 	}
 }
 
+static
+void
+ep_sample_event_pipe_callback(
+    const uint8_t* source_id,
+    unsigned long is_enabled,
+    uint8_t level,
+    uint64_t match_any_keywords,
+    uint64_t match_all_keywords,
+    EventFilterDescriptor* filter_data,
+    void* callback_data)
+{
+    if (filter_data) {
+        ep_char8_t *filter_data_char = (ep_char8_t *)((uintptr_t)(filter_data->ptr));
+        const uint32_t filter_data_size = filter_data->size;
+        ep_char8_t *candidateKey = NULL;
+        size_t offset = 0;
+        const ep_char8_t *sampleProfilerIntervalMSKey = "SampleProfilerIntervalMS";
+
+        while (offset < filter_data_size) {
+            candidateKey = filter_data_char + offset;
+            if (strcmp(candidateKey, sampleProfilerIntervalMSKey) == 0) {
+                ep_sample_profiler_set_sampling_rate(strtoull(candidateKey + 25, NULL, 10) * 1000000);
```

Comment:
We should check that strtoull returns a value greater than 0. If someone passes in a bad value that cannot be parsed it would return 0 and the sample thread would sample continuously and busy lock the app.

Reply:
We should probably even add logic to make sure the value falls within some accepted limits, so maybe a value between 1000 samples/second (current default and not sure we want to go lower than that with current sampling solution) and some lower limit that makes sense. You might want to sample quite infrequently in some cases, so not sure what the upper limit could be, so maybe we should just cap it at some sane value, like 1 sample/second, since its still a sample profiler. If you would like to get even more infrequent samples, then maybe it should be handled by a custom provider and not the sample profiler.

Reply:
Agreed that keeping a sane range is a good idea.  For example, ETW will not allow you to sample any more than every 0.125 seconds if I recall correctly.  I'm not super concerned about the other side of the range because there isn't a risk of affecting the availability of the app - there is a risk of not getting data.  I think it's probably safe to allow any large value, though at a certain point it doesn't really provide much value for the scenarios that we are currently tracking.